examples/webserver/Makefile

```diff
@@ -7,239 +7,44 @@
 ifeq ($(strip $(MICROKIT_SDK)),)
 $(error MICROKIT_SDK must be specified)
 endif
-MICROKIT_SDK:=$(abspath ${MICROKIT_SDK})
+override MICROKIT_SDK:=$(abspath ${MICROKIT_SDK})
 
 ifeq ($(strip $(WEBSITE_DIR)),)
 $(error "WEBSITE_DIR must be specified: this is the filesystem path of the directory containing the website's files")
 endif
 
 ifeq ($(strip $(LIBGCC)),)
-LIBGCC:=$(dir $(realpath $(shell aarch64-none-elf-gcc --print-file-name libgcc.a)))
+export LIBGCC:=$(dir $(realpath $(shell aarch64-none-elf-gcc --print-file-name libgcc.a)))
 endif
 
-ifeq ($(strip $(LIBMATH)),)
-LIBMATH:=$(dir $(realpath $(shell aarch64-none-elf-gcc --print-file-name libm.a)))
-endif
-
-LIBC := $(dir $(realpath $(shell aarch64-none-elf-gcc --print-file-name libc.a)))
-
-MICROKIT_CONFIG ?= debug
-BUILD_DIR ?= $(abspath build)
-MICROKIT_BOARD := odroidc4
-ETHERNET_CONFIG_INCLUDE ?= ethernet_config
-CPU := cortex-a55
-
-TOOLCHAIN := clang
-CC := clang
-LD := ld.lld
-TARGET := aarch64-none-elf
-MICROKIT_TOOL ?= $(MICROKIT_SDK)/bin/microkit
-
-BOARD_DIR := $(MICROKIT_SDK)/board/$(MICROKIT_BOARD)/$(MICROKIT_CONFIG)
-LIONSOS ?= ../..
-SDDF := $(LIONSOS)/dep/sddf
+export MICROKIT_CONFIG ?= debug
+export BUILD_DIR ?= $(abspath build)
+export MICROKIT_BOARD := odroidc4
+export CONFIG_INCLUDE ?= $(abspath config)
+export WEBSERVER_SRC_DIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+export LIONSOS := $(abspath ../..)
 
-LWIP=$(SDDF)/network/ipstacks/lwip/src
-LIBNFS=$(LIONSOS)/dep/libnfs
-NFS=$(LIONSOS)/components/fs/nfs
-MUSL=$(LIONSOS)/dep/musllibc
 
-IMAGES := timer.elf ethernet.elf micropython.elf nfs.elf eth_copy.elf eth_virt_rx.elf eth_virt_tx.elf uart_driver.elf serial_virt_rx.elf serial_virt_tx.elf
-CFLAGS := \
-	-mtune=$(CPU) \
-	-mstrict-align \
-	-ffreestanding \
-	-O3 \
-	-Wall \
-	-Wno-unused-function \
-	-I$(BOARD_DIR)/include \
-	-target $(TARGET) \
-	-DBOARD_$(MICROKIT_BOARD) \
-	-I$(SDDF)/include
-LDFLAGS := -L$(BOARD_DIR)/lib
-LIBS := -lmicrokit -Tmicrokit.ld
 
 IMAGE_FILE := $(BUILD_DIR)/webserver.img
 REPORT_FILE := $(BUILD_DIR)/report.txt
+nproc:= $(shell sh -c "if test -f /proc/cpuinfo; then grep processor /proc/cpuinfo | wc -l; else echo 2; fi")
```

Comment:
Hmm. Not sure about this, it should be up to the user to decide how many threads Make should start. Anyways, we can sort that out later.